ComfortZoneApp/app/src/main/java/com/example/comfortzone/utils/LocationUtil.java

```diff
@@ -72,4 +80,35 @@ public void onComplete(@NonNull Task<Location> task) {
                     }
                 });
     }
+
+    public static Observable<Object> getLocationObservable(Activity activity) {
```

Comment:
ohhh you did it! very nice! so now I think you can try the flatmap to make this cleaner